compiler/optimizer/LoopCanonicalizer.cpp

```diff
@@ -1999,6 +2074,38 @@ void TR_LoopCanonicalizer::eliminateRedundantInductionVariablesFromLoop(TR_Regio
                      }
                   }
                }
+            static bool disableTrivialArrayOperationCheck = (NULL != feGetEnv("TR_DisableTrivialArrayOperationCheck"));
+            bool isTrivialArrayOperation = false;
+            if (!disableTrivialArrayOperationCheck && !naturalLoop->isAcyclic())
+               {
+               if (trace())
+                  traceMsg(comp(), "Starting the trivial array operation check");
+
+               TR::Node *derivedInductionVarNode = NULL;
+               TR_RegionStructure::SubNodeList::iterator nodeIt = naturalLoop->begin();
+               TR_Structure *pred = toStructureSubGraphNode(*nodeIt)->getStructure();
+               for (TR::Block *derivedInductionVarBlock = pred->getEntryBlock();
+                    !derivedInductionVarNode && !derivedInductionVarBlock->isEndBlock();
+                    derivedInductionVarBlock = derivedInductionVarBlock->getNextBlock())
+                  {
+                  for (TR::TreeTopIterator treeTopIt(derivedInductionVarBlock->getEntry(), comp());
+                       !derivedInductionVarNode && treeTopIt != derivedInductionVarBlock->getExit();
+                       treeTopIt.stepForward())
+                     {
+                     if (treeTopIt.currentNode()->getSymbol() == derivedInductionVar->getSymbol())
+                        derivedInductionVarNode = treeTopIt.currentNode();
+                     }
+                  }
+
+               TR_ASSERT_FATAL_WITH_NODE(derivedInductionVarNode, derivedInductionVarNode != NULL, "Unable to find node containing the derived induction variable");
+               TR::NodeChecklist visited(comp());
+               isTrivialArrayOperation = trivialArrayOperationCheck(visited, derivedInductionVarNode);
+               if (isTrivialArrayOperation)
+                  isTrivialArrayOperation = performTransformation(comp(), "%sSkipping induction variable elimination for %d as it is involved in a trivial array operation.\n", OPT_DETAILS, derivedInductionVar->getReferenceNumber());
```

Comment:
This trace needs to include a loop number or something as well to make it easier to follow the log without the opt being traced.

Reply:
Apologies for the delay. The loop number is already being printed to the trace log here https://github.com/eclipse/omr/blob/4d32dfda3c999f4e1c27b7dcca73ab27c1989a0f/compiler/optimizer/LoopCanonicalizer.cpp#L1956